scripts/quests/otherAreas/RQ7_The_Basics.lua

```diff
@@ -0,0 +1,205 @@
+                        return quest:event(94) -- Quest starting event.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [94] = function(player, csid, option, npc)
+                    if option == 85 then -- Accept quest option.
+                        player:setCharVar("Quest[4][5]DayCompleted", 0)  -- Delete previous quest (The clue) variables.
+                        npcUtil.giveKeyItem(player, xi.ki.MHAURAN_COUSCOUS) -- Give Key Item to player.
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepeted.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 0
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            -- TODO: Find information about the ferry free ride. NPC involved and number of times it allows for free rides.
+            -- KNOWN: It isnt mandatory to take the ferry.
+
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(95) -- Not delivered the Key Item yet.
+                end,
+            },
+        },
+
+        [xi.zone.SELBINA] =
+        {
+            ['Valgeir'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(106) -- Deliver Key Item.
+                end,
+            },
+
+            onEventFinish =
+            {
+                [106] = function(player, csid, option, npc)
+                    npcUtil.giveItem(player, xi.items.BAKED_POPOTO)
+                    player:delKeyItem(xi.ki.MHAURAN_COUSCOUS)
+                    player:messageSpecial(selbinaID.text.KEYITEM_OBTAINED + 1, xi.ki.MHAURAN_COUSCOUS)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+    },
+
+    -- Section: Finish quest.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(95) -- Commentary.
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, {xi.items.BAKED_POPOTO}) then
+                        return quest:event(96) -- Quest completed.
+                    end
+                end,
+
+            },
+
+            onEventFinish =
+            {
+                [96] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:tradeComplete()
+                        player:addExp(2000)
+                        quest:setVar(player, 'CommentaryValgeir', 1)
```

Comment:
I'm not sure if topaz has a bit check/set function but you could make  it like remove these vars completely and at the npc for optional dialog have it check for a 'Option' var
If not isBitSet(quest:getVar(player, 'Option'), 1, 1) then
return cs
OnFinish....
setBit(quest:getVar(player, 'Option'), 1,1)
All quest vars will auto clear at quest compete also

Reply:
Adding to this, the functions are declared in `scripts/globals/interaction/container.lua` under Variable helper functions